polyply/src/graph_utils.py

```diff
@@ -150,3 +150,25 @@ def _compute_path_length_cartesian(molecule, mol_idx, path, nonbond_matrix):
                                                       node_from,
                                                       node_to)[0]
     return path_length
+
+def compute_avg_step_length(molecule, mol_idx, start, nonbond_matrix, stop=None):
+    """
+    Computes an average step length on a path.
+    """
+    # compute the shortest path between the ends in graph space
+    if stop:
+        end_to_end_path = nx.algorithms.shortest_path(molecule,
+                                                      source=start,
+                                                      target=stop)
+        end_to_end_path = list(zip(end_to_end_path[:-1], end_to_end_path[1:]))
+    else:
+        end_to_end_path = list(nx.dfs_edges(molecule, source=start))
```

Comment:
This looks a little like a kludge.
Maybe single_source_shortest_path? 
https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.shortest_paths.unweighted.single_source_shortest_path.html#networkx.algorithms.shortest_paths.unweighted.single_source_shortest_path

And then get the longest one:
```python
paths = single_source_shortest_path(...)
longest_path = max(paths.values(), key=len)
```